src/listing.rs

```diff
@@ -256,12 +276,14 @@ pub fn directory_listing(
                     Err(_) => None,
                 };
 
+                let size = WalkDir::new(entry.path()).into_iter().count();
```

Comment:
Thinking about it, we might be able to just do `.take(UPPER_LIMIT).count()` instead of `.count()` to shortcut execution time. So, we might have `UPPER_LIMIT` be something like 10001 and if it's more than 10000 we have the Impl Display for EntrySize print `>10000` or something along those lines.

Reply:
The other option would have been to add an actual timeout, but I'm not really sure how to best do that in a non-async context.